modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/utilities/NetCDFUtilities.java

```diff
@@ -453,7 +452,7 @@ public static int getDimensionLength(Variable var, final int dimensionIndex) {
      */
     public static int getRawDataType(final VariableIF variable) {
         VariableDS ds = (VariableDS) variable;
-        final DataType type = ds.getOriginalDataType();
+        final DataType type = ds.getDataType();
```

Comment:
I think I know why they did this the old way: integers are treated as categorical data. I will detail this in an email.